routing/pathfind_test.go

```diff
@@ -3130,3 +3131,186 @@ func dbFindPath(graph *channeldb.ChannelGraph,
 
 	return route, err
 }
+
+// TestBlindedRouteConstruction tests creation of a blinded route with the
+// following topology:
+//
+// A -- B -- C (introduction point) - D (blinded) - E (blinded).
+func TestBlindedRouteConstruction(t *testing.T) {
+	t.Parallel()
+
+	var (
+		// We need valid pubkeys for the blinded portion of our route,
+		// so we just produce all of our pubkeys in the same way.
+		_, alicePk = btcec.PrivKeyFromBytes([]byte{1})
+		_, bobPk   = btcec.PrivKeyFromBytes([]byte{2})
+		_, carolPk = btcec.PrivKeyFromBytes([]byte{3})
+		_, davePk  = btcec.PrivKeyFromBytes([]byte{4})
+		_, evePk   = btcec.PrivKeyFromBytes([]byte{5})
+
+		_, blindingPk = btcec.PrivKeyFromBytes([]byte{9})
+
+		sourceVertex         = route.NewVertex(alicePk)
+		currentHeight uint32 = 100
+
+		metadata           = []byte{1, 2, 3}
+		carolEncryptedData = []byte{4, 5, 6}
+		daveEncryptedData  = []byte{7, 8, 9}
+		eveEncryptedData   = []byte{10, 11, 12}
+
+		blindedPath = &sphinx.BlindedPath{
+			IntroductionPoint: carolPk,
+			BlindingPoint:     blindingPk,
+			BlindedHops: []*sphinx.BlindedHopInfo{
+				{
+					// The first payload should contain
+					// carol's blinded pubkey, but we don't
+					// need it here.
+					Payload: carolEncryptedData,
+				},
+				{
+					NodePub: davePk,
+					Payload: daveEncryptedData,
+				},
+				{
+					NodePub: evePk,
+					Payload: eveEncryptedData,
+				},
+			},
+		}
+
+		// Create final hop parameters for amount = 110 and delta = 50.
+		totalAmt   lnwire.MilliSatoshi = 110
+		finalDelta uint16              = 50
+
+		finalHopParams = finalHopParams{
+			amt:         totalAmt,
+			totalAmt:    totalAmt,
+			cltvDelta:   finalDelta,
+			records:     nil,
+			paymentAddr: nil,
+			metadata:    metadata,
+		}
+
+		// Create channel edges for the unblinded portion of our
+		// route. Proportional fees are omitted for easy test
+		// calculations, but non-zero base fees ensure our fee is
+		// still accounted for. Feature vectors are left empty because
+		// we don't need them for intermediate nodes.
+		aliceBobEdge = &channeldb.CachedEdgePolicy{
+			ChannelID: 1,
+			// We won't actually use this timelock / fee (since
+			// it's the sender's outbound channel), but we include
+			// it with different values so that the test will trip
+			// up if we were to include the fee/delay.
+			TimeLockDelta: 10,
+			FeeBaseMSat:   50,
+			ToNodePubKey: func() route.Vertex {
+				return route.NewVertex(bobPk)
+			},
+			ToNodeFeatures: tlvFeatures,
+		}
+
+		bobCarolEdge = &channeldb.CachedEdgePolicy{
+			ChannelID:     2,
+			TimeLockDelta: 15,
+			FeeBaseMSat:   20,
+			ToNodePubKey: func() route.Vertex {
+				return route.NewVertex(carolPk)
+			},
+			ToNodeFeatures: tlvFeatures,
+		}
+
+		// Create edges for the blinded portion of the route with no
+		// fee values, as would be the case when we construct a route
+		// from blinded-route-generated hop hints.
+		carolDaveEdge = &channeldb.CachedEdgePolicy{
```

Comment:
Shouldn't this edge include the aggregate conditions for the blinded route, mimicking the behaviour of findPath where we construct the edgepolicies via toRouteHints()